test/EndToEndTests/Tests/Client/AsynchronousTests/AsynchronousQueryTests.cs

```diff
@@ -807,6 +809,22 @@ public async Task Linq_ProjectPropertiesFromEntityWithConditionalNullCheckOnExpa
             Assert.Equal(-10, computer.ComputerDetail.ComputerDetailId);
         }
 
+        [Fact]
+        public async Task Linq_ProjectPropertiesFromNestedExpandedEntityToADifferentTargetTypeFromTheSource()
+        {
+            var context = this.CreateWrappedContext<DefaultContainer>().Context;
+            var query = context.ComputerDetail.Where(c => c.ComputerDetailId == -10)
+                .Select(c => new Computer
+                {
+                    ComputerId = c.Computer.ComputerId,

```

Comment:
Could you add complex properties and navigation properties? Like can we have complex properties and navigation properties within the select because I think in the issue that's what the customer was trying.

Reply:
were they navigation properties?

Reply:
@ElizabethOkerio in the customer's sample code, the `comuni` entity was accessing the `comuni.Provincia` property's properties. This is more likely a navigation property since it has an ID: `comuni.Provincia.ID`. The class structure that they shared also show a foreign key from one entity to the other.

Reply:
Do we support selecting complex and navigation properties? If we do then I'll recommend you add the complex and navigation properties to this test.

Reply:
@ElizabethOkerio the customer did confirm that they were using navigation properties. That said, I have added a test that references complex properties. I have also found out that complex properties are not affected by this specific issue because they use a different code path that did not trigger the call that led to the exception.